TUnit.Core/DataSources/DataSourceProcessor.cs

```diff
@@ -269,7 +276,10 @@ public static class DataSourceProcessor
             
             if (items.Count > 0)
             {
-                yield return items.ToArray();
+                // Optimize: Use pre-sized array instead of ToArray()
```

Comment:
fyi, `items.ToArray()` will pre-size automatically, since `items` is an `ICollection<>`

[source](https://source.dot.net/#System.Linq/System/Linq/ToCollection.cs,45)